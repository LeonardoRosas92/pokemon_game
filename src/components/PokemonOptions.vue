<template>
  <div class="options-container">
    <ul>
        <button v-for="pokemon in pokemons" :key="pokemon.id" 
            @click="$emit('selection', pokemon.id)">
            {{pokemon.name}}
        </button>
    </ul>
  </div>
</template>

<script>
export default {
    name: "PokemonOptions",
    props:{
        pokemons:{
            type: Array,
            required: true
        }
    }
}
</script>

<style scoped>
/* Pokemon Options */
.options-container{
    margin-top: 30px;
    display: flex;
    justify-content: center;
}
ul {
    list-style-type: none;
    padding: 0;
    
}
button {
    background-color: white;
    border-radius: 5px;
    cursor: pointer;
    margin-bottom: 10px;
    width: 250px;
    height: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 20px;
    box-shadow: 1px 1px 1px 0px rgba(227,225,225,0.75);
}

/* button:hover {
    background-color: #E54222;
    font-weight: bolder;
    color: #1C4CBD;
} */

</style>