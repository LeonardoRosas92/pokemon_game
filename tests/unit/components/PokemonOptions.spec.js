import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions'
import { pokemons } from '../mocks/pokemons.mock'
describe('PokemonOptions component', () => {

    let wrapper;

    beforeEach( () => {
        wrapper = shallowMount (PokemonOptions,{
            props:{
                pokemons: pokemons
            }
        })
    });

    test('Debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Debe mostrar las 4 opciones correcta', () => {
        const btns = wrapper.findAll('button');
        expect(btns.length).toBe(4);
        expect(btns[0].text()).toBe('bulbasaur');
        expect(btns[1].text()).toBe('ivysaur');
        expect(btns[2].text()).toBe('venusaur');
        expect(btns[3].text()).toBe('charmander');
    })

})