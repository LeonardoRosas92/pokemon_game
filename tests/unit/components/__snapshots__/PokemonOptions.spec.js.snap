// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component Debe de emitir "seleccion" con sus respectivos parametros al hacer click 1`] = `
<div class="options-container">
  <ul><button>bulbasaur</button><button>ivysaur</button><button>venusaur</button><button>charmander</button></ul>
</div>
`;

exports[`PokemonOptions component Debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul><button>bulbasaur</button><button>ivysaur</button><button>venusaur</button><button>charmander</button></ul>
</div>
`;
